airflow/example_dags/example_dataset_event_extra.py

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example DAG to demonstrate annotating a dataset event with extra information.
+"""
+
+from __future__ import annotations
+
+import datetime
+
+from airflow.datasets import Dataset
+from airflow.datasets.metadata import Metadata
+from airflow.models.dag import DAG
+from airflow.operators.bash import BashOperator
+
+ds = Dataset("s3://output/1.txt")
+
+with DAG(
+    dag_id="produce_with_yield",
```

Comment:
I wonder if "produce" is enough context for the dag_id. Maybe this instead, better matches the existing example dag_ids?

```suggestion
    dag_id="dataset_produces_with_yield",
```

Reply:
*Dataset produces* seems a bit weird too… We don’t need to include *produce* in the ID at all. I’ll change this to something more descriptive (and longer).